apps/admin/src/pages/sponsors/update-sponsor-form.tsx

```diff
@@ -0,0 +1,171 @@
+    mutate({ id: sponsor.id, ...values })
+  };
+
+  // fill array with all valid hackathonIds
+  const { data: hackathons } = trpc.hackathons.getAll.useQuery();
+
+  // useEffect to call getHackathonIds on component mount
+  useEffect(() => {
+    if (hackathons) {
+      const fetchedHackathonIds = hackathons.map((hackathon) => hackathon.id) || [];
+      setHackathonIds(fetchedHackathonIds);
+    }
+  }, [hackathons]);
+
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit(onSubmit)}
+        className="flex flex-col justify-center space-y-6"
+      >
+        <FormField
+          control={form.control}
+          name="name"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Name</FormLabel>
+              <FormControl>
+                <Input placeholder="Name" {...field} />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={form.control}
+          name="website"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Website</FormLabel>
+              <FormControl>
+                <Input placeholder="website" {...field} />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={form.control}
+          name="logo"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Logo</FormLabel>
+              <FormControl>
+                <Input placeholder="logo" {...field} />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField
+          control={form.control}
+          name="tier"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>Tier</FormLabel>
+              <FormControl>
+                <Input placeholder="tier" {...field} />
+              </FormControl>
+              <FormMessage />
+            </FormItem>
+          )}
+        />
+        <FormField 
+          control={form.control}
+          name="hackathonId"
+          render={({ field }) => (
+            <FormItem>
+              <FormLabel>HackathonId</FormLabel>
+                <Select defaultValue={String(field.value)} onValueChange={(selectedValue) => {
+                  // Convert the selected value to a number
+                  const numericValue = parseInt(selectedValue, 10);
+                  // Now you can use `numericValue` as a number
+                  field.onChange(numericValue);
+                }} >
+                  <FormControl>
+                    <SelectTrigger>
+                      <SelectValue placeholder="Select Hackathon Id" />
+                    </SelectTrigger>
+                  </FormControl>
+                  <SelectContent>
+                    {hackathonIds.map((id) => (
```

Comment:
You could probably refactor this to be something like this when just using the `hackathons` array:

```ts
{hackathons.map((hackathon) => {
  return (
    <SelectItem key={hackathon.id} value={String(hackathon.id)}>
      {hackathon.name}
    </SelectItem>
  )
})}
```

Reply:
refactored it biggest of bros